tidy3d/components/grid/grid_spec.py

```diff
@@ -199,6 +271,52 @@ def _make_coords_initial(
         return center - size / 2 + np.arange(num_cells + 1) * dl_snapped
 
 
+class CustomGridBoundaries(GridSpec1d):
+    """Custom 1D grid supplied as a list of grid cell boundary coordinates.
+
+    Example
+    -------
+    >>> grid_1d = CustomGridCoords(boundaries=[-0.2, 0.0, 0.2, 0.4, 0.5, 0.6, 0.7])
+    """
+
+    coords: Coords1D = pd.Field(
+        ...,
+        title="Grid Boundary Coordinates",
+        description="An array of grid boundariy coordinates.",
+        units=MICROMETER,
+    )
+
+    def _make_coords_initial(
+        self,
+        axis: Axis,
+        structures: List[StructureType],
+        **kwargs,
+    ) -> Coords1D:
+        """Customized 1D coords to be used as grid boundaries.
+
+        Parameters
+        ----------
+        axis : Axis
+            Axis of this direction.
+        structures : List[StructureType]
+            List of structures present in simulation, the first one being the simulation domain.
+        *kwargs
```

Comment:
these kwargs aren't used?

Reply:
It seems a historic thing (the same in UniformGrid). I think we kept it to make function signature consistent across subclasses, otherwise previously pylint will complain.

Reply:
if so maybe we should remove from the docstring then

Reply:
OK, I'll also remove it from UniformGrid